lib/3scale/backend/storage_helpers.rb

```diff
@@ -284,16 +280,35 @@ def cfg_unix_path_handler(options)
             options
           end
 
+          # Note that we can set reconnect_attempts to >= 0 because we use
+          # a monkey patch which implements a workaround for this issue
+          # that shows that there might be duplicated transactions when
+          # there's a timeout: https://github.com/redis/redis-rb/issues/668
+          # We should investigate if there are edge cases that can lead to
+          # duplicated commands because of this setting.
+          def cfg_reconnect_handler(options)
+            options[:reconnect_wait_seconds] = ThreeScale::Backend.listener? ? 0 : 5
+            options[:reconnect_attempts] = if ThreeScale::Backend.listener?
+                                             1
+                                           else
+                                             ThreeScale::Backend.configuration.redis.async ? Float::INFINITY : 1
+                                           end
```

Comment:
```suggestion
            options[:reconnect_attempts] = ThreeScale::Backend.listener? ? 1 : 2**30
```

Your original approach to make both sync and async wait a long time seemed good to me. So it is fine for me to set just a big value, not necessarily `INFINITY`.

But if you like to keep original sync behavior to `1` that's also fine. Everything looks good either way. Thank you!